src/support/TaskCommandRunnerFactory.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { CommandLineArgs, CommandNotSupportedError} from './commandLineBuilder';
+import { CommandRunner, ICommandRunnerFactory, Like, normalizeCommandResponseLike, PromiseCommandResponse, StreamingCommandRunner, VoidCommandResponse } from '../support/CommandRunner';
+
+interface TaskCommandRunnerOptions { //specifies name and types of properties and methods an object must have to be a instance of the interface
+    taskName: string;
+    workspaceFolder?: vscode.WorkspaceFolder; //legit a workspace folder
+    cwd?: string; //current working directory for the purpose of executing commands or accessing files
+    alwaysRunNew?: boolean;
+    rejectOnError?: boolean;
+    focus?: boolean;
+    env?: never; // Environment is not needed and should not be used, because VSCode adds it already (due to using `ExtensionContext.environmentVariableCollection`)
+}
+
+export class TaskCommandRunnerFactory implements ICommandRunnerFactory {
+    public constructor(private readonly options: TaskCommandRunnerOptions) { //takes the interface properties and makes them available in 'options'
+    }
+
+    
+    public getCommandRunner(): CommandRunner { 
+        return async <T>(commandResponseLike: Like<PromiseCommandResponse<T> | VoidCommandResponse>) => {
+            const commandResponse = await normalizeCommandResponseLike(commandResponseLike);
+            const output = await executeAsTask(this.options, commandResponse.command, commandResponse.args);
+            return output as T;
+        };
+    }
+
+    public getStreamingCommandRunner(): StreamingCommandRunner {
+        throw new CommandNotSupportedError('Streaming commands are not supported for task runners');
+    }
+}
+
+async function executeAsTask(options: TaskCommandRunnerOptions, command: string, args?: CommandLineArgs): Promise<void> {
+    const shellExecutionOptions = { cwd: options.cwd || options.workspaceFolder?.uri?.fsPath || os.homedir() };
+
+    const shellExecution = args ?
+        new vscode.ShellExecution(command, args, shellExecutionOptions) : // Command is the process, and args contains arguments
+        new vscode.ShellExecution(command, shellExecutionOptions); // Command is the full command line
+    const task = new vscode.Task(
+        { type: 'shell' },
+        options.workspaceFolder ?? vscode.TaskScope.Workspace,
+        options.taskName,
+        'Containers',
```

Comment:
Is this the correct Task source name? I would check the API Reference here to understand what each parameter does: https://code.visualstudio.com/api/references/vscode-api#Task